packages/cli-kit/src/private/node/ui/components/List.tsx

```diff
@@ -25,23 +34,57 @@ const List: FunctionComponent<ListProps> = ({
   color,
   bullet = DOT,
 }): JSX.Element => {
+  function isCustomListItem(item: TokenItem<InlineToken> | CustomListItem): item is CustomListItem {
+    return (item as CustomListItem).item !== undefined
+  }
```

Comment:
why do we have to support `TokenItem<InlineToken>` if we cast and override return type 😅 

edit: actually I see, we can replace this with your new `ListItem` type. I think we can probably do better with the typing here, but I won't hold things up. Let me know if you want to pair on this!